M3-3416 [Longview] Plan Details page

johnwcallahan opened the pull request:
## Description

This PR contains the new Longview "Plan Details" page.

In this PR:

- New [SDK] method: GET `/longview/subscriptions`
- New `LongviewSubscription` interface
- New `longviewSubscription` factory for generating test data (via factory.ts)
- Updates to `accountSettings.container.ts`to include the `updateAccountSettings` request.
- `<LongviewPlans />` component (with sub components in the same file)
- Tests ~(still in progress)~ done for now

Mobile styling is... fine. I made the table non-responsive, so you'll need to horizontally scroll on smaller viewports to see the entire table.

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers

To run new unit tests: `yarn test packages/manager/src/features/Longview/LongviewLanding/LongviewPlans.test.tsx` ~(still in progress)~ done for now

To test functionality, go to `http://localhost:3000/longview/plan-details` and change your subscription. Simulate error state by request-blocking `*longview*` and `*settings*`.

martinmckenna commented:
agreed with all of Jared's points - otherwise LGTM

Jskobos commented:
@johnwcallahan one more request: we're going to need the longview/subscriptions data (as well as the accountSettings) in the longview/clients page. Would it make sense to move your requests up to LongviewLanding and pass that data down to both tabs?

johnwcallahan commented:
Thanks for the feedback. @Jskobos addressing your points: 

> We should handle for Managed customers more explicitly. We could disable the form for them with a message, for example.

cc @alioso 
I chatted with Jay about this. We're going to hide the table and show a message:

![Screen Shot 2019-11-07 at 3 14 10 PM](https://user-images.githubusercontent.com/16911484/68426671-9b360980-0176-11ea-9a6f-d6a73caeab25.png)

> Pressing the submit button with the current plan selected doesn't do anything & there's no feedback. Should disable that one somehow, like we do for the Linode resize flow

Also chatted with Jay about this. The submit button has been disabled if you've selected the plan that is already on your account.

> Weird bug: generate a few errors then navigate to /account/settings...you get toasts for the errors you've just seen inline in the form

This is super interesting. This is happening because upon failure to update your account settings, we get an error in Redux under `accountSettings.error.update`. `<GlobalSettings />` blindly shows toasts for `update` errors upon every render: https://github.com/linode/manager/blob/2150815ecf8aec6e3743e42186ab2391ccecb4bd/packages/manager/src/features/Account/GlobalSettings.tsx#L130

My feeling is that this should be fixed separately, in that component. Let me know your thoughts.
Another solution is that `<LongviewPlans />` could clear out `accountSettings` Redux errors on unmount, but that feels wrong to me.

> Would it make sense to move your requests up to LongviewLanding and pass that data down to both tabs?
Yep, done. Though I kept the accountSettings container where it is, because I also needed `isManaged`. We can hook up the parent to `accountSettings` as well.
